libs/sdk-backend-tiger/src/backend/index.ts

```diff
@@ -193,8 +308,312 @@ export class TigerBackend implements IAnalyticalBackend {
+                    dataSourceId: string,
+                    generateLogicalModelRequest: GenerateLdmRequest,
+                ) => {
+                    return await this.authApiCall(async (sdk) => {
+                        try {
+                            return await sdk.axios.post(
+                                `/api/v1/actions/dataSources/${dataSourceId}/generateLogicalModel`,
+                                generateLogicalModelRequest,
+                            );
+                        } catch (error) {
+                            throw convertApiError(error);
+                        }
+                    });
+                },
+                scanDataSource: async (dataSourceId: string, scanRequest: ScanRequest) => {
+                    return await this.authApiCall(async (sdk) => {
+                        return await sdk.axios
+                            .post(`/api/v1/actions/dataSources/${dataSourceId}/scan`, scanRequest)
+                            .then((res: AxiosResponse) => {
+                                return Promise.resolve(res?.data);
+                            })
+                            .catch((err) => {
+                                return Promise.reject(`scan error=${JSON.stringify(err.response.data)}`);
+                            });
+                    });
+                },
+                publishPdm: async (dataSourceId: string, declarativePdm: DeclarativePdm) => {
+                    return await this.authApiCall(async (sdk) => {
+                        return await sdk.declarativeLayout
+                            .setPdmLayout({
+                                dataSourceId,
+                                declarativePdm,
+                            })
+                            .then(
+                                () => {
+                                    return {
+                                        status: "success",
+                                    };
+                                },
+                                (error) => {
+                                    return Promise.reject(error);
+                                },
+                            );
+                    });
+                },
+                createDemoDataSource: async (sampleDataSource?: JsonApiDataSourceInDocument) => {
+                    return await this.authApiCall(async (sdk) => {
+                        try {
+                            const result = await sdk.entities.createEntityDataSources({
+                                jsonApiDataSourceInDocument: sampleDataSource
+                                    ? { ...sampleDataSource }
+                                    : {
+                                          data: {
+                                              attributes: {
+                                                  name: "demo-ds",
+                                                  url: "jdbc:postgresql://localhost:5432/demo",
```

Comment:
Do we need this default? Should this always come from outside instead? I am a bit "triggered" by the connection string hardcoded here.

Reply:
Yeah, I was not sure about this either. I removed that default and use it directly from where it is called. It makes more sense now 👍

Reply:
Yeah, I was not sure about this either. I removed that default and use it directly from where it is called. It makes more sense now 👍